lib/types.ts

```diff
@@ -274,8 +274,22 @@ export class TIface extends TType {
  * Defines an optional property on an interface.
  */
 export function opt(typeSpec: TypeSpec): TOptional { return new TOptional(parseSpec(typeSpec)); }
-export class TOptional {
-  constructor(public ttype: TType) {}
+export class TOptional extends TType {
+  constructor(public ttype: TType) {
+    super();
+  }
+
+  public getChecker(suite: ITypeSuite, strict: boolean): CheckerFunc {
+    const itemChecker = this.ttype.getChecker(suite, strict);
+    return (value: any, ctx: IContext) => {
+      if (value === undefined) {
+        return true;
+      }
+
+      const failMsg: string = (this.ttype as TName)._failMsg;
```

Comment:
I think it's better to avoid this hack, keep `_failMsg` private, and simply `return itemChecker(value, ctx)` when value is not `undefined`. It's risky, both in failing when `ttype` doesn't actually contain a `_failMsg` member, and in that constructing an error message by adding "or omitted" to whatever the message happens to be, might not produce a human-friendly message.

I think simply keeping the original message is perfectly sufficient. E.g. for `[number, string?]`, checking `[5]` or `[5, "asdf"]` would pass, while checking `[5,6]` would yield `value[1] is not a string`. I think that's totally fine, no?